File: rust/axum/src/common/helper/jwt.rs
use std::env;
use std::time::{SystemTime, UNIX_EPOCH, Duration};
use jsonwebtoken::{encode, Header, EncodingKey, Algorithm};
use crate::modules::auth::interfaces::JwtPayload;

#[derive(Debug)]
pub struct JwtReturn {
	pub token: String,
	pub expires_in: SystemTime,
}

pub fn generate_jwt(payload: JwtPayload) -> Result<JwtReturn, Box<dyn std::error::Error>> {
	let secret = env::var("JWT_SECRET").unwrap_or_else(|_| "default_secret".to_string());
	let expires_in = env::var("JWT_EXPIRES_IN")
		.unwrap_or_else(|_| "86400".to_string())
		.parse::<u64>()
		.unwrap_or(86400);

	let exp = SystemTime::now() + Duration::from_secs(expires_in);
	let exp_timestamp = exp.duration_since(UNIX_EPOCH)?.as_secs();

	let claims = serde_json::json!({
		"cpf": payload.cpf,
		"email": payload.email,
		"id": payload.id,
		"role": payload.role,
		"exp": exp_timestamp
	});

	let token = encode(
		&Header::new(Algorithm::HS256),
		&claims,
		&EncodingKey::from_secret(secret.as_bytes()),
	)?;

	Ok(JwtReturn {
		token,
		expires_in: exp,
	})
}

File: go/fiber_orm/internal/common/helper/jwt.go
package helper

import (
	"os"
	"strconv"
	"time"

	auth "tcc/internal/modules/auth/interfaces"

	"github.com/golang-jwt/jwt/v5"
)

var SECRET = os.Getenv("JWT_SECRET")
var EXPIRES_IN = os.Getenv("JWT_EXPIRES_IN")

type Return struct {
	Token string
	ExpiresIn time.Time
}

func GenerateJWT(payload auth.JwtPayload) (*Return, error){
	expires_in, err := strconv.Atoi(EXPIRES_IN)

	if err != nil {
		expires_in = 86400
	}

	var exp = time.Now().UTC().Add(time.Second * time.Duration(expires_in))
	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
		"cpf": payload.CPF,
		"email": payload.Email,
		"id": payload.ID,
		"role": payload.Role,
		"exp": exp.Unix(),
	})

	tokenString, err := token.SignedString([]byte(SECRET))

	if err != nil {
    return nil, err
	}	

	return &Return{
		Token: tokenString,
		ExpiresIn: exp,
	}, nil
}